packages/language-service/src/codefixes/fix_missing_import.ts

```diff
@@ -0,0 +1,165 @@
+    // In the future, we could handle multiple matches as additional quick fixes.
+    const allPossibleDirectives = checker.getPotentialTemplateDirectives(templateInfo.component);
+    const matchingDirectives =
+        getDirectiveMatchesForElementTag(missingElement, allPossibleDirectives);
+    if (matchingDirectives.size === 0) {
+      return [];
+    }
+    const bestMatch = [...matchingDirectives][0];
+    const originalImportedSymbolName = bestMatch.tsSymbol.name;
+
+    // Get possible trait imports corresponding to the recommended directive. Only use the
+    // compiler's best import; in the future, we could suggest multiple imports if they exist.
+    const potentialImports = checker.getPotentialImportsFor(bestMatch, importOn);
+    if (potentialImports.length === 0) {
+      return [];
+    }
+    const potentialImport = potentialImports[0];
+
+    // The trait might already be imported, possibly under a different name. If so, determine the
+    // local name of the imported trait.
+    const allImports =
+        findAllMatchingNodes(importOn.getSourceFile(), {filter: ts.isImportDeclaration});
+    const existingImportName: string|null =
+        hasImport(allImports, potentialImport.symbolName, potentialImport.moduleSpecifier);
+
+    // If the trait has not already been imported, we need to insert the new import.
+    let fileImportChange = {
+      fileName: importOn.getSourceFile().fileName,
+      textChanges: new Array<ts.TextChange>(),
+    };
+    if (existingImportName === null) {
+      const existingImportDeclaration = allImports.find(
+          decl => moduleSpecifierHasPath(decl.moduleSpecifier, potentialImport.moduleSpecifier));
+      const importName = nonCollidingImportName(allImports, potentialImport.symbolName);
+
+      if (existingImportDeclaration !== undefined) {
+        // Update an existing import declaration.
+        const bindings = existingImportDeclaration.importClause?.namedBindings;
+        if (bindings === undefined || ts.isNamespaceImport(bindings)) {
+          // This should be impossible, as a namespace imports are already existing for any symbol.
+          return [];
+        }
+        let span = {start: bindings.getStart(), length: bindings.getWidth()};
+        const updatedBindings = updateNamedImport(bindings, potentialImport.symbolName, importName);
+        const importString = printNode(updatedBindings, importOn.getSourceFile());
+        fileImportChange.textChanges.push({span, newText: importString});
+      } else {
+        // Generate a new import declaration, and insert it at the top of the file.
+        const newImport =
+            generateImport(potentialImport.symbolName, importName, potentialImport.moduleSpecifier);
+        const importString = printNode(newImport, importOn.getSourceFile()) + '\n';
+        fileImportChange.textChanges.push({span: {start: 0, length: 0}, newText: importString});
+      }
```

Comment:
```suggestion
    fileImportChange.textChanges.push(someHelper(...))
```
Would it make sense to extract this chunk to a helper function that returns the additional text change?